File: src/main/java/io/lenses/alerting/plugin/AlertingPlugin.java
package io.lenses.alerting.plugin;

import io.lenses.alerting.plugin.util.Try;

import java.util.Map;

interface AlertingPlugin {
    /**
     * Returns AlertingService for valid configuration (depends on the alerting service class).
     * @param config service configuration
     * @return alerting service or failure
     */
    Try<AlertingService> init(Map<String, String> config);
}

File: src/main/scala/scalaapi/AlertingPlugin.scala
package io.lenses.alerting.plugin.scalaapi

import java.util

import io.lenses.alerting.plugin.{AlertingPlugin => JAlertingPlugin}
import io.lenses.alerting.plugin.util.{Try => JTry}

import scala.util.Try

/**
  * simple alerting plugin wrapper giving Scala users a bit more type information
  */
class AlertingPlugin(alertingPlugin: JAlertingPlugin) {

  def init(config: util.Map[String, String]): Try[AlertingService] = for {
    init <- Try { alertingPlugin.init(config) }
  } yield x
//    Try { new AlertingService(alertingPlugin.init(config)) }
}
